pandas/core/arrays/base.py

```diff
@@ -787,7 +797,12 @@ def unique(self):
         uniques = unique(self.astype(object))
         return self._from_sequence(uniques, dtype=self.dtype)
 
-    def searchsorted(self, value, side="left", sorter=None):
+    def searchsorted(
+        self,
+        value: ArrayLike,
+        side: Optional[Literal["left", "right"]] = "left",
```

Comment:
`Optional` not needed. within the method body, `side` will be a `str` and not `None`

Reply:
> I don't see that. `searchsorted` calls `numpy.searchsorted` which allows `None` as an argument.

it's optional. None raises `TypeError`

```
>>> 
>>> np.searchsorted([1,2,3,4,5], 3)
2
>>> 
>>> np.searchsorted([1,2,3,4,5], 3, side='right')
3
>>> 
>>> np.searchsorted([1,2,3,4,5], 3, side=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 5, in searchsorted
  File "/home/simon/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 1343, in searchsorted
    return _wrapfunc(a, 'searchsorted', v, side=side, sorter=sorter)
  File "/home/simon/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 55, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/home/simon/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 44, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
TypeError: search side must be str, not NoneType
>>> 
```